sympy/core/expr.py

```diff
@@ -826,7 +827,11 @@ def _eval_interval(self, x, a, b):
         value = B - A
 
         if a.is_comparable and b.is_comparable:
-            singularities = list(solveset(self.cancel().as_numer_denom()[1], x))
+            if a < b:
```

Comment:
Your shortcut conditional expression looks better than mine. 
I never used it. I will try to use it in the future. Thanks.